### PR TITLE
Directly base this on ubuntu/bionic

### LINKED ISSUE #12: use ubuntu:bionic
This can now use `FROM ubuntu:bionic` instead of gap-container
https://github.com/gap-system/gap-docker-base/blob/042b826c0727ad88f4ca4652e23a0059966de6a9/Dockerfile#L1

### PR DESCRIPTION
**Pull request by fingolfin**
Resolves #12

### CODE REVIEW

**Comment by fingolfin** [comment]
In addition, I also fixed the CXSC download URL, and modified the list of system packages being installed, putting it into alphabetical order and one item per line, which should make future maintenance nice.

**Comment by fingolfin** [comment]
Also, I have another PR ready which puts the various versions into env vars, and then updates Singular to the latest; I'd rather not do too much in a single PR.

**Comment by fingolfin** [comment]
It also seems that Singular in here cannot find Flint:
```
checking for FLINT >= 2.4... problem
Sorry, your FLINT version is too old. Disabling.
configure: WARNING: Unable to find FLINT (which is strongly recommended) on your machine: please use --with-flint=PATH_TO_DIR_CONTAINING_LIB_AND_INCLUDE (see also ./configure --help if you do not understand what we are talking about)
```

Of course that in turn might be due to Flint's `make install` not really installing everything?!? It seems to only install headers, not the `libflint` binary. I guess one of the dangers of just using the `flint` master branch; this `Dockerfile` really should be using a tag or a specific revision

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,52 @@
...
+            automake \
+            build-essential \
+            cmake \
+            g++ \
+            gcc \
+            gcc-multilib \
+            git \
+            libboost-dev \
+            libcdd-dev \
+            libglpk-dev \
```
Comment (fingolfin) [inline Dockerfile]: @alex-konovalov Why is this here? Perhaps for @sebasguts GLPKInterface? But that doesn't ship with GAP, does it?

File: Dockerfile [hunk 2]
```diff
@@ -1,22 +1,52 @@
...
+            autogen \
+            automake \
+            build-essential \
+            cmake \
+            g++ \
+            gcc \
+            gcc-multilib \
+            git \
+            libboost-dev \
+            libcdd-dev \
```
Comment (fingolfin) [inline Dockerfile:21]: @alex-konovalov @sebasguts Any idea why this is here? Best I can think of is that polymake uses cdd; but unless one wants to compile polymake, that shouldn't matter... Can I remove it?

File: Dockerfile (hunk 1 above)
Comment (fingolfin) [reply Dockerfile]: So: Can I remove it?

File: Dockerfile (hunk 2 above)
Comment (fingolfin) [reply Dockerfile:21]: Perhaps 4ti2 (which we build from source later on) needs it?

**Comment by fingolfin** [comment]
I fixed the build issues with Singular, and also fixed the PARI URL. This version of the Dockerfile now builds just fine for me. So from my point of view, this is ready to merge.

File: Dockerfile (hunk 2 above)
Comment (sebasguts) [reply Dockerfile:21]: I think for polymake, but I am not sure.

File: Dockerfile (hunk 1 above)
Comment (sebasguts) [reply Dockerfile]: I think it can be removed, yes.

**Review by sebasguts — APPROVED** [review]

**Comment by fingolfin** [comment]
Also added a few more packages which are needed by gap-docker, gap-docker-master, gap-docker-stable-4.10. @alex-konovalov can this be merged now? And then an update be pushed to Docker hub?

**Comment by olexandr-konovalov** [comment]
@fingolfin did you add all what apt-get installs at https://github.com/gap-system/gap-docker-master/blob/master/Dockerfile ?

**Comment by fingolfin** [comment]
@alex-konovalov yes, all dependencies have been carefully transferred. It would be good to get this merged, and then further cleanup other containers and/or delete those we don't need anymore.

**Comment by olexandr-konovalov** [comment]
@fingolfin ok thanks. I guess if any of the dependencies is missing, we will discover that :) and if not, then we will get rid of some unused dependency as a side benefit :)

**Comment by fingolfin** [comment]
I was careful about this when writing this PR, and just now went over it again together with @zickgraf, so I am pretty confident. But of course mistakes still haben.

I think Fabian may submit some PRs for the other repos soon, to adjust them to this PR being merged.
